dispatch/dispatch.go

```diff
@@ -161,20 +167,44 @@ func (d *Dispatcher) run(it provider.AlertIterator) {
 				return
 			}
 
-			level.Debug(d.logger).Log("msg", "Received alert", "alert", alert)
+			// this block is wrapped in a function to make sure that the span
```

Comment:
Let's extract all this logic into a method like `dispatch` or something? This is what I was thinking:

```go
	for {
		select {
		case alert, ok := <-it.Next():
			if !ok {
				// Iterator exhausted for some reason.
				if err := it.Err(); err != nil {
					level.Error(d.logger).Log("msg", "Error on alert update", "err", err)
				}
				return
			}
			d.dispatch(alert)
```